components/core/src/clp_s/SchemaReader.hpp

```diff
@@ -166,6 +168,7 @@ class SchemaReader {
     std::unordered_map<int32_t, int32_t> m_local_id_to_global_id;
 
     JsonSerializer m_json_serializer;
+    bool m_should_marshal_records{true};
```

Comment:
Do we need to initialize it here since it is initialized in the constructor?

Reply:
I think the pattern we've been using recently is to do sensible default initialization this way in case someone adds a new constructor/changes existing constructors without accounting for it. We don't need to initialize this here, but I'd prefer to keep it.

Reply:
Yep, what Devin said based on this clang-tidy check: https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-default-member-init.html